internal/handlers/common/update_array_operators.go

```diff
@@ -315,3 +315,100 @@ func processAddToSetArrayUpdateExpression(doc, update *types.Document) (bool, er
 
 	return changed, nil
 }
+
+// processPullAllArrayUpdateExpression changes document according to $pullAll array update operator.
+func processPullAllArrayUpdateExpression(doc, update *types.Document) (bool, error) {
+	var changed bool
+
+	iter := update.Iterator()
+
+	for {
+		key, pullAllValueRaw, err := iter.Next()
+		if err != nil {
+			if errors.Is(err, iterator.ErrIteratorDone) {
+				break
+			}
+
+			return false, lazyerrors.Error(err)
+		}
+
+		path, err := types.NewPathFromString(key)
+		if err != nil {
+			return false, lazyerrors.Error(err)
+		}
+
+		// If the path does not exist, create a new array and set it.
+		if !doc.HasByPath(path) {
+			if err = doc.SetByPath(path, types.MakeArray(1)); err != nil {
+				return false, commonerrors.NewWriteErrorMsg(
+					commonerrors.ErrUnsuitableValueType,
+					err.Error(),
+				)
+			}
+		}
+
+		val, err := doc.GetByPath(path)
+		if err != nil {
+			return false, err
```

Comment:
Hmm, after passing `HasByPath`, this is rather a lazy error or a panic (as if everything is fine, this line doesn't seem to be reachable).